back/engines/commercial/insights/app/services/insights/category_assignments_service.rb

```diff
@@ -125,5 +126,20 @@ def touch_views_of(categories)
     def update_counts_of(categories)
       categories.each { |cat| cat.update({ inputs_count: CategoryAssignment.where(category_id: cat.id).size })}
     end
+
+    # Sets a processed_flag for all the inputs in the view corresponding to the categories
+    def set_processed(inputs, categories)
+      processed_flag_attributes = inputs.to_a.product(categories) # yields all pairs of input x category
```

Comment:
Here if I could be sure all the categories and inputs live in the same view I'd only create one flag, but I'm not sure, so here is that. (Maybe useful for topic import)

Reply:
Maybe something like that?
```
view_ids = categories.pluck(:view_id).uniq
```
then do the cross-product between the inputs and the view ids.